packages/react/src/useStructuralMemo.ts

```diff
@@ -0,0 +1,18 @@
+import deepEqual from "deep-equal";
+import { useRef } from "react";
+
+/**
+ * Memoize and ensure a stable identity on a given value as long as it remains the same, structurally.
+ */
+export const useStructuralMemo = <T>(value: T): T | undefined => {
```

Comment:
Any chance we could import this or something similar from urql? I feel like they probably need to so something similar internally such that we could avoid shipping extra JS kbs to the client